packages/calcite-components/src/components/filter/filter.tsx

```diff
@@ -219,16 +235,16 @@ export class Filter
 
   clear = (): void => {
     this.value = "";
-    this.filter("", true);
+    this.filterDebounced("", true);
     this.setFocus();
   };
 
-  updateFiltered(filtered: any[], emit = false): void {
-    this.filteredItems.length = 0;
-    this.filteredItems = this.filteredItems.concat(filtered);
+  updateFiltered(filtered: object[], emit = false, callback?: () => void): void {
+    this.filteredItems = [...filtered];
```

Comment:
is a shallow copy good enough here, considering `filteredItems` are objects?

Reply:
Thats a good question. It was a shallow copy previously right?

Reply:
I think a shallow copy is fine here since we weren't cloning before.

Reply:
Yeah I looked at the code and we definitely don't need a deep copy and probably don't even need a shallow one. `filteredItems` is readonly and our [filter util](https://github.com/esri/calcite-components/blob/dris0000/filter-method/packages/calcite-components/src/utils/filter.ts) uses [`Array.prototype.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)  which already creates a shallow copy. So we can do:

```ts
updateFiltered(filtered: object[], emit = false, callback?: () => void): void {
    this.filteredItems = filtered;
```